src/app/api/users/users.service.ts

```diff
@@ -33,14 +32,31 @@ class UsersService extends BaseService {
 
     // Filter out companies where isPlaceholder is true if companies.data is not null
     const filteredCompanies = companies.data ? companies.data.filter((company) => !company.isPlaceholder) : []
-    const clientsWithCompanyData = clients.data?.map((client) => {
-      const companyForClient = filteredCompanies.find((company) => company.id === client.companyId)
-      return { ...client, fallbackColor: companyForClient?.fallbackColor }
-    })
+    const clientsWithCompanyData =
+      clients.data?.map((client) => {
+        const companyForClient = filteredCompanies.find((company) => company.id === client.companyId)
+        return { ...client, fallbackColor: companyForClient?.fallbackColor }
+      }) || []
 
     return { internalUsers: ius.data, clients: clientsWithCompanyData, companies: filteredCompanies }
   }
 
+  /**
+   * Service to filter users starting with a given keyword
+   * @param keyword
+   * @returns
+   */
+  async getFilteredUsersStartingWith(keyword: string) {
+    const filterByKeyword = (users: FilterableUser[]) => filterUsersByKeyword(users, keyword)
+
+    const { internalUsers, clients, companies } = await this.getGroupedUsers(100_000)
```

Comment:
what is `100_000`. That's not a number right? Im unsure how this is working.

A more realistic limit might be 500.

Reply:
This is ES2021 syntactic sugar for separating larger numbers so they are easier to read. I don't see much JS/TS devs using this but from my time doing Ruby I find it to be really elegant for dev readability @makeitraina 

```js
const regularMiliion = 1000000
const readableMiliion = 1_000_000
```

Ref: https://www.30secondsofcode.org/js/s/numeric-separator/#:~:text=Numeric%20separators%20are%20a%20lesser,to%20make%20them%20more%20readable.

Reply:
Very interesting! Good to know. This is resolved